clash-lib/src/Clash/Util.hs

```diff
@@ -268,7 +268,9 @@ reportTimeDiff start end =
     (Clock.UTCTime (toEnum 0) (fromRational (toRational diff)))
  where
   diff = Clock.diffUTCTime start end
-  fmt  | diff >= 3600
+  fmt  | diff >= 86400
+       = "%-Dd%-Hh%-Mm%-S%03Qs"
```

Comment:
According to the [docs](https://hackage.haskell.org/package/time-1.12/docs/Data-Time-Format.html#v:formatTime)
`%D` for `Day` (which includes `UTCTime`) means: `%m/%d/%y`

Reply:
You might hope to use `%d` directly and be good up to a month, but that doesn't work either.
Because `0 :: UTCTime` is 1858-11-17 00:00:00 UTC

Ideally we should be using `NominalDiffTime` instead of `UTCTime`, but sadly that's only supported in `time >= 1.9`.
And we can only use `time >= 1.9` starting with `ghc >= 8.8`. (`time` itself can compile on older GHC versions, but since it's also a dependency of `ghc` we have to use the `time` version GHC ships with)
Which makes it no good for clash-1.4, and also not for `master` as it supports ghc-8.6.

Reply:
We use `utcDiffTime` which returns a `NominalDiffTime` where the `D` modifier denotes Days (of week). Note that that’s not day-of-the-week.